lib/elixir/lib/kernel.ex

```diff
@@ -4998,6 +4998,53 @@ defmodule Kernel do
     Macro.escape(NaiveDateTime.from_iso8601!(string))
   end
 
+  @doc ~S"""
+  Handles the sigil `~U` for UTC date times.
+
+  The lower case `~u` variant does not exist as interpolation
+  and escape characters are not useful for datetime sigils.
+
```

Comment:
We should have a paragraph saying that Z or 0000 offsets are required, not having one or having the wrong one will raise.